src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -154,12 +157,12 @@ public IRecordingDescriptor startRecording(
                     Optional<IRecordingDescriptor> previous =
                             getDescriptorByName(connection, recordingName);
                     if (previous.isPresent()) {
-                        if (!restart) {
+                        if (!restartRecording) {
                             throw new IllegalArgumentException(
                                     String.format(
                                             "Recording with name \"%s\" already exists",
                                             recordingName));
-                        } else {
+                        } else if (isRecordingStopped(previous.get())) {
```

Comment:
I'm wondering what happens if the previous recording exists, but `replace == stopped`. According to the original issue, apparently nothing should happen, meaning the recording should not be stopped and restarted. That logic should still be applied here.

Reply:
Ok, so when recording exist and is stopped do nothing?

Reply:
I think that if the previous recording exist and  is stopped it should be restarted and if it is not stopped do nothing

Reply:
Let's say the user put `replace` == 'stopped' and didn't specify `restart` and the recording already exists and is still running.

The code would still go and attempt to start a new recording from my understanding. You are right about the case where the recording is `STOPPED`, but I think if it's `RUNNING` is a different issue.

Reply:
if its running do nothing, that's what I understood

Reply:
Not sure if I understand the problem completely, @andrewazores what was your intention with the `stopped` parameter.

Reply:
`?replaced=stopped` should behave like:

> I think the best thing to do here is actually just ignore the request on any targets that already have such a recording running, since starting a new parallel active recording in the same running state doesn't actually capture any additional data. If the recording doesn't exist, it should be created and started. If it exists but has been stopped, it should be restarted. If it is already running just do nothing.

ie. the "replace this recording" logic should only apply to recordings that exist and are `STOPPED` - so close that recording and start a new one again with the same name etc. If the recording is not `STOPPED`, ie it's `RUNNING`, then do not replace it - in other words, leave it running and do nothing. If there is no such recording then create a new one with the given settings.

If the user has specified both `?replace=something&restart=another` then I'm not sure what to do. `restart` is the older and now-deprecated form of the same parameter so I think in this case we can just respond with an `HTTP 400`.

Reply:
> also if both parameters are specified by the user i.e ?replace=something&restart=another replace should take preference.. or we should add a code that handles this specific scenario like HTTP 400?

I think either way of handling it would be okay. Either reject requests with a 400 if both parameters are specified, or just ignore the old/deprecated parameter if the newer form is specified. Both seem to make sense, as long as we document it (HTTP_API.md).

Reply:
OK, I think I am ignoring the old/deprecated parameter when the newer form is specified. Please review if you have time and let me know @andrewazores @maxcao13